libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -610,15 +611,14 @@ open class GossipRouter(
         enqueueIwant(peer, messageIds)
     }
 
-    private fun iDontWant(msg: PubsubMessage, receivedFrom: PeerHandler) {
+    private fun iDontWant(msg: PubsubMessage) {
```

Comment:
TBH I would leave the excluded peer. We are saving one IDONTWANT message for free. 
```suggestion
    private fun iDontWant(msg: PubsubMessage?) {
```